batch/batch/cloud/gcp/worker/worker_api.py

```diff
@@ -128,3 +138,111 @@ async def close(self):
 
     def __str__(self):
         return f'project={self.project} zone={self.zone}'
+
+
+ContainerCredentials = Dict[str, aiogoogle.GoogleServiceAccountCredentials]
+
+
+class GoogleHailMetadataServer(HailMetadataServer[GCPUserCredentials, ContainerCredentials]):
+    def __init__(self, project: str, http_session: httpx.ClientSession):
+        super().__init__()
+        self._project = project
+        self._metadata_server_client = aiogoogle.GoogleMetadataServerClient(http_session)
+
+    def _create_container_credentials(self, default_credentials: GCPUserCredentials):
```

Comment:
Are these *default_credentials*? Aren't they the credentials of the service account of the user of this job?

Reply:
The name for this variable has since changed but yes to both :). In GCE, the service account assigned to the machine is aliased to `default`. So it's the default credentials for the machine, but also the only one.